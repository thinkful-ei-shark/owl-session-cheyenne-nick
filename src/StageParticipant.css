.stage-participant img {
    margin-top: 100px;
}