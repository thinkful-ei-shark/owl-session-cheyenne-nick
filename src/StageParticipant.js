import React, { Component } from 'react';

class StageParticipant extends Component {
    render() {
        return (
            <div className="stage-participant item group-column">
                <div className="stage-settings item group-column">
                    <h3 className='stage-name'>this.props.name</h3>
                    <div className="audio-bar"></div>
                    <div className="icons group-row">
                        <button className="speaker">🔊</button>
                        <button className="mag-glass">🔍</button>
                    </div>
                </div>
                <img src={this.props.img.src} alt={this.props.name} className="profile-pic item" />
            </div>
        )
    }
} 

export default StageParticipant;