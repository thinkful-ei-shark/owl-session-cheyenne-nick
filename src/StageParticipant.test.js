import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import StageParticipant from './StageParticipant'

describe('<StageParticipant />', () => {
    it('renders without crashing', () => {
        // Create a DOM element to render the component into
        const div = document.createElement('div');

        // Render the component
        // If something is wrong it will fail here
        ReactDOM.render(<StageParticipant key = {1} name = "name" avatar = 'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'/>, div);

        // Clean up
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const participant = renderer
            .create(<StageParticipant key = {1} name = "name" avatar = 'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'/>)
            .toJSON();
        
        expect(participant).toMatchSnapshot();
    });
});
