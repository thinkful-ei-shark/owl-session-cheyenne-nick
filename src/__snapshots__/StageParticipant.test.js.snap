// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StageParticipant /> renders the UI as expected 1`] = `
<div
  className="stage-participant item group-column"
>
  <div
    className="stage-settings item group-column"
  >
    <h3
      className="stage-name"
    >
      this.props.name
    </h3>
    <div
      className="audio-bar"
    />
    <div
      className="icons group-row"
    >
      <button
        className="speaker"
      >
        🔊
      </button>
      <button
        className="mag-glass"
      >
        🔍
      </button>
    </div>
  </div>
  <img
    alt="name"
    className="profile-pic item"
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
  />
</div>
`;
